src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala

```diff
@@ -0,0 +1,118 @@
+  override def requires: Plugins =
+    GatlingBasicInfoPlugin &&
+      GatlingReleasePlugin &&
+      GatlingPublishPlugin &&
+      Sonatype
+
+  trait GatlingSonatypeKeys {
+    val gatlingPublishToSonatype = settingKey[Boolean]("true to publish to sonatype")
+  }
+  object GatlingSonatypeKeys extends GatlingSonatypeKeys
+  object autoImport extends GatlingSonatypeKeys
+
+  import autoImport._
+
+  override def projectSettings: Seq[Def.Setting[_]] = Seq(
+    gatlingPublishToSonatype := ensurePublishableVersion(version.value),
+    sonatypeProfileName := "io.gatling",
+    publishTo := {
+      if (gatlingPublishToSonatype.value) {
+        sonatypePublishTo.value
+      } else {
+        publishTo.value
+      }
+    },
+    sonatypeSessionName := s"[sbt-sonatype] ${githubPath.value} ${(ThisBuild / version).value}",
+    releasePublishArtifactsAction := {
+      if (gatlingPublishToSonatype.value) {
+        publishSigned.value
+      } else {
+        Keys.publish.value
+      }
+    },
+    gatlingReleasePublishStep := conditionalPublishStep
+  )
+
+  val conditionalPublishStep: ReleaseStep = { state: State =>
+    if (Project.extract(state).get(gatlingPublishToSonatype)) {
+      publishStep(state)
+    } else {
+      GatlingReleasePlugin.publishStep(state)
+    }
+  }
+
+  val publishStep: ReleaseStep = { state: State =>
+    /*
+     * Issues:
+     *  - sbt-sonatype plugin only declares commands (not tasks)
+     *  - sonatypeOpen command calls appendWithoutSession, and release version is reset to its -SNAPSHOT
+     *
+     * Workaround:
+     *  - retrieve sonatypePublishTo settings after applying sonatypeOpen command
+     *  - inject it to the state needed by publishSigned task
+     *  - call sonatypeClose command with full state from sonatypeOpen
+     */
+    state.log.info(s"Opening sonatype staging")
+    val sonatypeOpenState = releaseStepCommandAndRemaining("sonatypeOpen")(state)
+    val sonatypeTargetRepositoryProfileValue = sonatypeOpenState.getSetting(sonatypeTargetRepositoryProfile).get
+
+    val startStateWithSonatypeConf = reapply(
+      Project.extract(state).currentProject.referenced.flatMap { ref =>
+        ref / sonatypeTargetRepositoryProfile := sonatypeTargetRepositoryProfileValue
+      } ++
+        Seq(
+          sonatypeTargetRepositoryProfile := sonatypeTargetRepositoryProfileValue
+        ),
+      state
+    )
+
+    state.log.info("compile, package, sign and publish")
+    val endState = {
+      val extracted = Project.extract(startStateWithSonatypeConf)
+      extracted.runAggregated(
+        extracted.currentRef / releasePublishArtifactsAction,
+        startStateWithSonatypeConf
+      )
+    }
+
+    state.log.info("Closing sonatype staging")
+    Def.unit(releaseStepCommandAndRemaining("sonatypeClose")(sonatypeOpenState))
+    endState
+  }
+
+  def ensurePublishableVersion(str: String): Boolean = str.matches("""\d+\.\d+\.\d+(\.\d+)?(-M\d+)?""")
```

Comment:
we now accept an optional 4th component of the version number (can be used by components like gatling-grpc, same principle we already use for the Gradle plugin versions).